docs/proposals/scheduling/20240201-enbale-reservation-preempt.md

```diff
@@ -0,0 +1,181 @@
+
+#### Priority
+When reservation is created, its priority should be set as followed.
+- `spec.preemptionPolicy` is filled by user, describe whether a reservation can trigger preemption.
+- `scheduling.koordinator.sh/reservation-priority` is filled by user, describe reservation 's priority, only higher
+  priority reservations can preempt lower priority reservations.
+
+#### Preemption
+The reservation preemption still follows the existing Filter/PostFilter procedure and can be combined with job-level
+preemption mechanism.
+
+The preemption strategy of reservation is as followed:
+1. Only preemptionPolicy=PreemptLowerPriority reservation can trigger preempt, and only lower priority reservations can be preempted.
+
+2. Reservations can only preempt reservations.
+
+3. If only part of Reservation can be assigned successfully in preemption dry-run process, the preemption will not
+   really happen.
+
+4. If reservation is preempted, the bound pods should also be evicted.
+
+#### Response To Preempted
+Once a reservation is chosen to be evicted, it will follow the scheduler implemented eviction mechanism, support soft-eviction
+or delete according to the implementation. The only difference is that we need to check the evicted is pod or reservation.
+
+#### Extension Point
+
+##### Over All
+Generally we will extend the elastic quota plugin, and modify other plugins to support reservation preemption.
+The new\delta parts are:
+1. Enable Reserve pod to preempt.
+2. Enable Reserve pod to be preempted.
+3. patch evict label/annotation to Reservation.
+4. Register Reservation eventHandler for job-oversold plugin.
+
+##### PreFilter
+If pod is Reserve pod, as reservation is not associated to quota yet, so it is no need to do quota check for reservation.
+
+##### PostFilter
+We will maintain the existing implementation process, but with the following differences.
+
+1. If pod is Reserve pod, as reservation is not associated to quota yet, so it is no need to do quota check for reservation.
+2. Reserve pods only preempt reserve pods.
+
+### API
+#### Reservation
+We introduce some labels to describe reservation behaviour.
+- `pod-group.scheduling.sigs.k8s.io` is filled by user, describe reservation should be scheduled in batch.
+- `scheduling.koordinator.sh/soft-eviction` is filled by scheduler, indicate the reservation is preempted.
+- `spec.preemptionPolicy` is filled by user, describe whether a reservation can trigger preemption.
+- `scheduling.koordinator.sh/reservation-priority` is filled by user, describe reservation 's priority, only higher
+  priority reservations can preempt lower priority reservations.
+
+For example, there are two Reservations as followed. If Reservation1 failed to schedule due to resource not enough, then
+Reservation1 can preempt Reservation2. Because Reservation1 spec.PreemptionPolicy equals to PreemptLowerPriority and its
+priority is higher than Reservation2's priority.
+
+Reservation1
+```yaml
+spec:
```

Comment:
These examples are not complete enough. The necessary kind and apiversion are missing, and does this spec refer to reservation.spec or reservation.spec.template.spec?